src/Microsoft.SqlTools.ServiceLayer/QueryExecution/ResultSet.cs

```diff
@@ -70,25 +65,31 @@ public class ResultSet : IDisposable
         private readonly string outputFileName;
 
         /// <summary>
+        /// Row count to use in special scenarios where we want to override the number of rows.
+        /// </summary>
+        private long? rowCountOverride;
+
+        /// <summary>
         /// The special action which applied to this result set
         /// </summary>
         private readonly SpecialAction specialAction;
 
+        /// <summary>
+        /// Total number of bytes written to the file. Used to jump to end of the file for append
+        /// scenarios. Internal for unit test validation.
+        /// </summary>
+        internal long totalBytesWritten;
```

Comment:
I think this gets set to 0 automatically, but I would be in favor of explicitly setting it to 0.